File: KtorApi-WhiskerWatch/src/main/kotlin/com/example/dto/CalendarDto.kt
package com.example.dto

import kotlinx.serialization.Serializable

/**
 * Clase DTO Serializable para la creacion de calendarios
 */
@Serializable
data class CalendarCreateDto(
    var mapsId: String,
    var listTasks: MutableList<TaskCreateDto>
)

/**
 * Clase DTO Serializable para la creacion de tareas
 */
@Serializable
data class TaskCreateDto(
    var date: String,
    var task: String
)

File: Frontend-WhiskerWatch/src/app/models/calendar/calendar-dto/calendar-dto.ts
export class CalendarDto {
  public id: string;
  public mapsId: string;
  public listTasks: Array<TaskDto>;

  constructor() {
    this.id = '';
    this.mapsId = '';
    this.listTasks = [];
  }
}

export class TaskDto {
  public id: string;
  public date: string;
  public task: string;

  constructor() {
    this.id = '';
    this.date = '';
    this.task = '';
  }
}
